apps/api/src/services/programs.ts

```diff
@@ -36,25 +39,43 @@ export class ProgramsService {
       .from(majorSpecialization)
       .innerJoin(major, eq(majorSpecialization.id, major.id))
       .innerJoin(degree, eq(major.degreeId, degree.id));
+
+    if (query.majorId) {
+      majorBuilder.where(eq(major.id, query.majorId));
+    }
+
+    return majorBuilder;
   }
 
-  async getMinors() {
-    return this.db
+  async getMinors(query: z.infer<typeof minorQuerySchema>) {
+    const minorBuilder = this.db
```

Comment:
Same here as above. Note that if you included screenshots of these query parameters in action or found another way to hold yourself accountable to test this new feature, this bug would have been caught.